# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptSettings.cpp
# Commit: f441cb5e7e6e
# Full Hash: f441cb5e7e6e30d4a60b08ca99e56a29d3dc3a8b
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r cfae5efaae37 -r f441cb5e7e6e dom/script/ScriptSettings.cpp
--- a/dom/script/ScriptSettings.cpp	Mon Mar 31 19:49:02 2025 +0000
+++ b/dom/script/ScriptSettings.cpp	Mon Mar 31 19:49:02 2025 +0000
@@ -233,6 +233,13 @@
   return xpc::NativeGlobal(global);
 }
 
+WebTaskSchedulingState* GetWebTaskSchedulingState() {
+  if (const nsIGlobalObject* global = GetEntryGlobal()) {
+    return global->GetWebTaskSchedulingState();
+  }
+  return nullptr;
+}
+
 nsIPrincipal* GetWebIDLCallerPrincipal() {
   MOZ_ASSERT(NS_IsMainThread());
   ScriptSettingsStackEntry* entry = ScriptSettingsStack::EntryPoint();